libs/sdk-core/src/errors.rs

```diff
@@ -0,0 +1,74 @@
+use reqwest::StatusCode;
+use std::time::Duration;
+
+pub type SdkResult<T, E = ExternalSdkError> = anyhow::Result<T, E>;
+
+// TODO Rename to SdkError once existing SDKError struct is replaced
+/// Public error kinds returned by the SDK
+#[derive(Debug)]
+pub enum ExternalSdkError {
+    LnInvoiceError { err: String },
+
+    // TODO Generic type, contains all new and unknown errors
+    Generic,
+}
+
+/// Conversion from granular internal errors to external user-facing SDK Errors
+impl From<InternalSdkError> for ExternalSdkError {
+    fn from(value: InternalSdkError) -> Self {
+        match value {
+            InternalSdkError::LnInvoiceError(err) => Self::LnInvoiceError { err },
+            _ => Self::Generic,
+        }
+    }
+}
+
+pub(crate) type InternalSdkResult<T, E = InternalSdkError> = anyhow::Result<T, E>;
+
+/// Error kinds thrown internally by SDK modules
+#[derive(Debug)]
+pub(crate) enum InternalSdkError {
```

Comment:
I am not sure we need internal sdk error structure. Every module can define its own errors if needed but I don't think it should be in a central place.

Reply:
I got rid of the `InternalSdkError`.